sp/src/game/shared/mapbase/vscript_funcs_shared.cpp

```diff
@@ -560,16 +560,18 @@ bool CAnimEventTInstanceHelper::Set( void *p, const char *pszKey, ScriptVariant_
 
 	animevent_t *ani = ((animevent_t *)p);
 	if (FStrEq( pszKey, "event" ))
-		ani->event = variant;
+		return variant.AssignTo( &ani->event );
 	else if (FStrEq( pszKey, "options" ))
-		ani->options = variant;
+		// broken: return variant.AssignTo( &ani->options );
+		//         variant memory is freed afterwards
```

Comment:
Same for `SetOptions`. In `_set` the variant owns the memory but in `SetOptions` the parameter points to a ref counted squirrel string which can be freed and reused during the lifetime of `animevent_t`.

Since this `animevent_t` is stack allocated in `CBaseAnimating::DispatchAnimEvents()`, a new `options` string memory could somehow be handled in there.

Reply:
From what I understand then, this is only used from `CBaseAnimating::ScriptHookHandleAnimEvent()`.
My solution now allocates a fresh buffer to copy the string into (and frees it in `DispatchAnimEvents()`). A more optimal solution might try to use the Squirrel VM string by incrementing its refcount, but `ScriptVariant_t` is currently not setup to track the Squirrel VM strings separately, so I opted to not do that.